File: JeongGod/10week/programmers/42883.py
def solution(number, k):
    """
    내림차순으로 가면 최댓값이 나온다.
    1. n[-1] >= val
        일단 더한다.
    2. n[-1] < val
        n[-1]은 버리고
        1이 나올때 까지 계속 반복한다.
        그 다음에 val을 추가한다.
    """
    n = [number[0]]
    for val in number[1:]:
        while n and k > 0 and n[-1] < val:
            n.pop()
            k -= 1
        n.append(val)
    answer = "".join(n[:len(number)-k])
    
    return answer

File: jekyun-park/week10/programmers/42883.swift
//
//  main.swift
//  SwiftAlgorithms
//
//  Created by 박제균 on 2022/03/03.
//  Programmers > 탐욕법(Greedy) > 큰 수 만들기

import Foundation

func solution(_ number: String, _ k: Int) -> String {

    var k = k
    var stack: [String] = []

    for num in number {
        
        while !stack.isEmpty && Int(stack.last!)! < Int(exactly: num.wholeNumberValue!)! && k > 0 {
            stack.removeLast()
            k -= 1
        }
        
        stack.append(String(num))
    }
    
    // 남은 k개의 수에 대해
    while k > 0 {
        stack.removeLast()
        k -= 1
    }
    
    return stack.joined()
}


print(solution("1924", 2))
print(solution("1231234", 3))
print(solution("4177252841", 4))





